### ISSUE TITLE
[Bug] For Each and Lists Bugs

### ISSUE THREAD

**Issue by MerlinofMines** [MerlinofMines/EasyCommands#185]
**Describe the bug**
Using Lists along with For Each commands leads to really strange behavior.  It affects keyed values as well as regular values.

Seems like either the List combine, or the for each command, is using a variable reference instead of the resolved value.

**To Reproduce**
```
set myList to ['a','b','c']

set myMap to []

for each myValue in myList
  myMap+=[myValue]

print myList
print myMap
```

Output:
[c,c,c]

**Expected behavior**
[a,b,c]

**Additional context**

**Comment by MerlinofMines** [comment]
It's actually not For Each, this is clearly an issue with lists when using variable values.  The default behavior is not using the static value but rather a reference.

```
set myString to "someString"
set myMap to []

myMap+=[myString]

print myMap
set myString to "someOtherString"

print myMap
```

**Comment by MerlinofMines** [comment]
This was resolved by #187
